src/util/sighash.rs

```diff
@@ -128,6 +127,71 @@ pub enum SchnorrSigHashType {
     /// Reserved for future use, `#[non_exhaustive]` is not available with current MSRV
     Reserved = 0xFF,
 }
+serde_string_impl!(SchnorrSigHashType, "a SchnorrSigHashType data");
+
+impl SchnorrSigHashType {
+    /// Break the sighash flag into the "real" sighash flag and the ANYONECANPAY boolean
+    pub(crate) fn split_anyonecanpay_flag(self) -> (SchnorrSigHashType, bool) {
+        match self {
+            SchnorrSigHashType::Default => (SchnorrSigHashType::Default, false),
+            SchnorrSigHashType::All => (SchnorrSigHashType::All, false),
+            SchnorrSigHashType::None => (SchnorrSigHashType::None, false),
+            SchnorrSigHashType::Single => (SchnorrSigHashType::Single, false),
+            SchnorrSigHashType::AllPlusAnyoneCanPay => (SchnorrSigHashType::All, true),
+            SchnorrSigHashType::NonePlusAnyoneCanPay => (SchnorrSigHashType::None, true),
+            SchnorrSigHashType::SinglePlusAnyoneCanPay => (SchnorrSigHashType::Single, true),
+            SchnorrSigHashType::Reserved => (SchnorrSigHashType::Reserved, false),
```

Comment:
Not sure that `Reserved` will not be changed in the future and some of its variants may support `ANYONECANPAY`. May be the function signature should be updated to return Optional as a second parameter?

Reply:
As above, I think this is unrelated to this PR since these lines show up as a refactor (move code block) only. And as above I would not be able to explain the change, can I leave this one with you also?